File: C_Programming/Practise_Programs/PP_054.c
/*
==============================================================================
 Program     : Display Digits of a Number

 Description : Reads an integer from the user and displays each digit of the
               specified number starting from the least significant digit.

 Author      : Shubham Gadhe
==============================================================================
*/

#include<stdio.h>

/*
==============================================================================
 Function    : DisplayDigits

 Description : Displays each digit of the specified number from right to left.

 Parameters  : iNo - Number whose digits are to be displayed.

 Returns     : None

 Complexity  : Time  : O(d)
               Space : O(1)

 Note        : d represents the number of digits in the given number.
==============================================================================
*/
void DisplayDigits(int iNo)
{
    int iDigit = 0;

    // Display each digit from right to left.
    while(iNo > 0)
    {
        iDigit = iNo % 10;
        printf("%d\n", iDigit);
        iNo = iNo / 10;
    }  
}

int main()
{
    int iValue = 0; 
    
    printf("Enter Number : \n");
    scanf("%d", &iValue);

    DisplayDigits(iValue);
    
    return 0;
}

/*
==============================================================================
 Sample Input
------------------------------------------------------------------------------
 751

 Sample Output
------------------------------------------------------------------------------
 1
 5
 7
==============================================================================
*/

File: Java_Programming/Practise_Programs/PP_054.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_054
//
//  Description       : This program accepts a number from the user
//                      and displays uppercase alphabets at odd
//                      positions and lowercase alphabets at
//                      even positions.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 07/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class PP_054
{
    //////////////////////////////////////////////////////////////////
    //
    //  Function Name    : Display()
    //
    //  Description      : It is used to display uppercase
    //                     alphabets at odd positions and
    //                     lowercase alphabets at even
    //                     positions.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 07/08/2026
    //
    //////////////////////////////////////////////////////////////////

    public static void Display(int iNo)
    {
        int iCnt = 0;
        char ch1 = '\0';    // For Capital
        char ch2 = '\0';    // For Small

        for(iCnt = 1, ch1 = 'A', ch2 = 'a'; iCnt <= iNo; iCnt++, ch1++, ch2++)
        {
            if(iCnt % 2 == 0)
            {
                System.out.print(ch2 + "\t");
            }
            else
            {
                System.out.print(ch1 + " \t");
            }
        }

        System.out.println();
    }

    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the number of elements : ");
        iValue = sobj.nextInt();

        Display(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      5
//
//  Sample Output :
//      A    b    C    d    E
//
//////////////////////////////////////////////////////////////////